File: jar/src/main/java/jline/api/sn/SnHasMultiServer.java
package jline.api.sn;

import jline.lang.NetworkStruct;

public final class SnHasMultiServer {
    private SnHasMultiServer() {}

    public static boolean snHasMultiServer(NetworkStruct sn) {
        // Infinite servers are delays, not multiserver queues: counting them
        // made every model with a Delay read as multiserver.
        if (sn.nservers == null) {
            return false;
        }
        for (int i = 0; i < sn.nservers.getNumRows(); i++) {
            for (int j = 0; j < sn.nservers.getNumCols(); j++) {
                double c = sn.nservers.get(i, j);
                if (!Double.isInfinite(c) && c > 1) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: jar/src/main/kotlin/jline/api/sn/SnHasMultiServer.kt
package jline.api.sn

import jline.lang.NetworkStruct

fun snHasMultiServer(sn: NetworkStruct): Boolean {
    return sn.nservers.elementMax() > 1
}
/**
 * Stochastic network HasMultiServer algorithms
 */
@Suppress("unused")
class SnhasmultiserverAlgo {
    companion object {
        // Class documentation marker for Dokka
    }
}